plugins/woocommerce/src/StoreApi/Authentication.php

```diff
@@ -94,15 +110,25 @@ protected function is_preflight() {
 	/**
 	 * Checks if we're using a cart token to access the Store API.
 	 *
-	 * @param \WP_REST_Request $request Request object.
+	 * @param string $cart_token The cart token. This will be looked up in the request header if not provided.
 	 * @return boolean
 	 */
-	protected function has_valid_cart_token( \WP_REST_Request $request ) {
-		$cart_token = $request->get_header( 'Cart-Token' );
-
+	public function has_valid_cart_token( $cart_token = null ) {
+		if ( ! $cart_token ) {
+			$cart_token = $this->get_cart_token();
+		}
 		return $cart_token && JsonWebToken::validate( $cart_token, $this->get_cart_token_secret() );
 	}
```

Comment:
Yeah, let's kill this one..

Reply:
https://github.com/woocommerce/woocommerce/pull/55047/commits/84cf630021cd8724c4b437e29243433d8d95f364